README.md
# Blackjack SE
##### by Miguel Garcia

## Description
Blackjack Simple Edition (SE) is my take on the popular casino game, Blackjack.

The initial goal of this project was to create a simple web-based Blackjack game. This involved using a single deck of cards, one player, and only two game actions (Hit and Stay), along with a straightforward UI design.

During development, I decided to incorporate animations to enhance the gameplay experience. Specifically, I implemented card rendering animations so that cards would appear on the screen as if they were being dealt, rather than instantly rendering on the screen. This also included animating the card counter.

After successfully creating a functional prototype of the game, my enthusiasm to expand upon it grew. I'm currently in the process of adding more game features to bring it closer to the actual gameplay of Blackjack and working on implementing customizable settings to enhance the player's experience.

I enjoy taking on fun side projects, as they help me refine my skills. I particularly relished working on this project because it allowed me to learn more about the terminology and rules of the game of Blackjack.

Thank you, and I hope you enjoy playing this game as much as I enjoyed creating it.



CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0) and this project adheres to [Semantic Versioning](http://semserver.org/spec/v2.0.0.html).

### Types of Changes
- `Added` for new features.
- `Changed` for changes in existing functionality
- `Deprecated` for soon-to-be removed features.
- `Fixed` for any bug fixes
- `Security` in case of vulnerabilities

### Semantic Versioning
Given a version number MAJOR.MINOR.PATCH, increment the:

1. MAJOR version when you make incompatible API changes
2. MINOR version when you add functionality in a backwards compatible manner
3. Patch version when you make backwards compatible bug fixes

# Changes 
## [0.0.2] - 10/22/23
### Changed
- updated CSS structure
- added CSS for small mobile devices
- updated feature request file

## [0.0.1] - 09/06/2023
### Added
- two new deck styles to the images directory
- logic to handle dynamic deck style switching
- CHANGELOG file
- feature-request file
- README file
- background images
- CSS media query files



mock-data/player-hands.js
let h1 = {
    Hand: [
        { Value: "2", Weight: 2 },
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 13
        { Value: "K", Weight: 10 }, // 13
        { Value: "5", Weight: 5 }, // 18
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 19
        { Value: "4", Weight: 4 }, // 23
    ],
};

let h2 = {
    Hand: [
        { Value: "2", Weight: 2 },
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 13
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 14
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 15
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 16
        { Value: "6", Weight: 6 }, // 12
    ],
};

let h3 = {
    Hand: [
        { Value: "2", Weight: 2 },
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 13
    ],
};

let h4 = {
    Hand: [
        { Value: "2", Weight: 2 },
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 13
        { Value: "K", Weight: 10 }, // 13
        { Value: "A", Weight: { firstWeight: 1, secondWeight: 11 } }, // 14
        { Value: "Q", Weight: 10 }, // 24
    ],
};

mock-data/card-decks.js
// #region Player's dealt Blackjack scenarios
// Scenario: Player's dealt Blackjack, Dealer's dealt Blackjack
export const set1 = [
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // D: 21
    { Value: "J", Suit: "spades", Weight: 10 }, // P: 21
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Blackjack, Dealer's dealt Stands
export const set2 = [
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 17
    { Value: "J", Suit: "spades", Weight: 10 }, // P: 21
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Blackjack, Dealer hits twice and Stands
export const set3 = [
    { Value: "3", Suit: "spades", Weight: 3 }, // D: 17
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 14
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 12
    { Value: "J", Suit: "spades", Weight: 10 }, // P: 21
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]
// #endregion Player's dealt Blackjack scenarios

// #region Player's dealt Stands scenarios

// Scenario: Player's dealt Stands, Dealer's dealt Blackjack
export const set4 = [
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // D: 21
    { Value: "5", Suit: "spades", Weight: 5 }, // P: 16
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer's dealt Stands : Player > Dealer
export const set5 = [
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 17
    { Value: "7", Suit: "spades", Weight: 7 }, // P: 18
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer's dealt Stands : Player < Dealer
export const set6 = [
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 17
    { Value: "5", Suit: "spades", Weight: 5 }, // P: 16
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer's dealt Stands : Player === Dealer
export const set7 = [
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 17
    { Value: "6", Suit: "spades", Weight: 6 }, // P: 17
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer hits twice and hits Blackjack
export const set8 = [
    { Value: "5", Suit: "spades", Weight: 5 }, // D: 21
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // D: 16
    { Value: "5", Suit: "spades", Weight: 5 }, // D: 15
    { Value: "6", Suit: "spades", Weight: 6 }, // P: 17
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer hits twice : Player > Dealer 
export const set9 = [
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 18
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 16
    { Value: "5", Suit: "spades", Weight: 5 }, // D: 15
    { Value: "8", Suit: "spades", Weight: 8 }, // P: 19
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer hits twice : Player < Dealer 
export const set10 = [
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 17
    { Value: "3", Suit: "spades", Weight: 3 }, // D: 15
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 12
    { Value: "5", Suit: "spades", Weight: 5 }, // P: 16
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// Scenario: Player's dealt Stands, Dealer hits twice : Player === Dealer 
export const set11 = [
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 17
    { Value: "3", Suit: "spades", Weight: 3 }, // D: 15
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 12
    { Value: "6", Suit: "spades", Weight: 6 }, // P: 17
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// * Scenario: Player's dealt Stands, Dealer hits twice and Busts
export const set12 = [
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 22
    { Value: "3", Suit: "spades", Weight: 3 }, // D: 15
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 12
    { Value: "6", Suit: "spades", Weight: 6 }, // P: 17
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]

// * Scenario: Player's dealt Stands at 13, Dealer hits twice and Busts
export const set14 = [
    { Value: "10", Suit: "spades", Weight: 10 }, // D: 26
    { Value: "5", Suit: "spades", Weight: 5 }, // D: 16
    { Value: "9", Suit: "spades", Weight: 9 }, // D: 11
    { Value: "3", Suit: "spades", Weight: 3 }, // P: 13
    { Value: "2", Suit: "spades", Weight: 2 }, // D: 2
    { Value: "K", Suit: "spades", Weight: 10 }, // P: 10
]

// #endregion Player's dealt Stands scenarios

// Scenario: Player hits twice and Busts
export const set15 = [
    { Value: "K", Suit: "spades", Weight: 10 }, // P: 22
    { Value: "6", Suit: "spades", Weight: 6 }, // P: 12
    { Value: "7", Suit: "spades", Weight: 7 }, // D: 17
    { Value: "5", Suit: "spades", Weight: 5 }, // P: 16
    { Value: "J", Suit: "spades", Weight: 10 }, // D: 10
    { Value: "A", Suit: "spades", Weight: { firstWeight: 1, secondWeight: 11 } }, // P: 11
]